cli/sql.go

```diff
@@ -74,11 +75,47 @@ http://www.cockroachlabs.com/docs/
 `)
 }
 
+// handleSet handles the \set command and changes client-side flags.
+func handleSet(cmd []string, confirm *bool) int {
+	if len(cmd) != 2 {
+		fmt.Fprintf(os.Stderr, "Usage:\n \\set <option>\n")
+		return cliNextLine
+	}
+
+	switch cmd[1] {
+	case `confirm`:
+		*confirm = true
+	case `noconfirm`:
+		*confirm = false
+	default:
+		fmt.Fprintf(os.Stderr, "unrecognized \\set option: %s\n", cmd[1])
+	}
+	return cliNextLine
+}
+
+// confirmStatement checks whether the statement string contains
+// "dangerous" statements and asks a confirmation to the user if
+// necessary.
+func confirmStatement(confirm bool, stmt string) bool {
+	stmt = strings.ToUpper(stmt)
+	if confirm && (strings.Contains(stmt, "DROP ")) {
```

Comment:
I think false positives are worse than false negatives (yes, data loss is worse than being annoyed by confirmation prompts, but we'll never be able to add a confirmation prompt to *all* potentially-destructive commands). How likely is it that a user will type or paste a multi-statement sequence including a DROP without meaning to execute it?